druid-shell/src/platform/x11/window.rs

```diff
@@ -505,32 +607,47 @@ impl Window {
             count: 0,
             button: MouseButton::None,
         };
-        if let Ok(mut handler) = self.handler.try_borrow_mut() {
-            handler.mouse_move(&mouse_event);
-        } else {
-            log::warn!("Window::handle_motion_notify - handler already borrowed");
+        match self.handler.try_borrow_mut() {
+            Ok(mut handler) => {
+                handler.mouse_move(&mouse_event);
+                Ok(())
+            }
+            Err(err) => Err(Error::BorrowError(format!(
+                "Window::handle_motion_notify handle: {}",
+                err
+            ))),
         }
     }
 
-    pub fn handle_destroy_notify(&self, _destroy_notify: &xcb::DestroyNotifyEvent) {
-        if let Ok(mut handler) = self.handler.try_borrow_mut() {
-            handler.destroy();
-        } else {
-            log::warn!("Window::handle_destroy_notify - handler already borrowed");
+    pub fn handle_client_message(
+        &self,
+        client_message: &xcb::ClientMessageEvent,
+    ) -> Result<(), Error> {
```

Comment:
That is correct right now. The return type is `Result` because I wanted to make the X11 event handling API uniform. Future extensions to `handle_client_message` might return an error and `Application` is ready for it because it expects all these handlers to return a `Result`.

Reply:
That is correct right now. The return type is `Result` because I wanted to make the X11 event handling API uniform. Future extensions to `handle_client_message` might return an error and `Application` is ready for it because it expects all these handlers to return a `Result`.